src/pkg/p2p/preheat/models/policy/policy.go

```diff
@@ -118,6 +118,19 @@ type Trigger struct {
 	} `json:"trigger_setting,omitempty"`
 }
 
+// ValidatePreheatPolicy validate preheat policy
+func (s *Schema) ValidatePreheatPolicy() error {
+	// currently only validate cron string of preheat policy
+	if s.Trigger != nil && s.Trigger.Type == TriggerTypeScheduled && len(s.Trigger.Settings.Cron) > 0 {
+		isValid, err := utils.ValidateCronString(s.Trigger.Settings.Cron)
+		if !isValid || err != nil {
```

Comment:
```
if err != nil {
 return err --- this is 500 error
}
if !isValid {
 return BadRequest -- this is 400 error
}
```

Reply:
actually，when the first field of cron is not 0, like "1 0 0 1 1 *", `utils.ValidateCronString` will return `false` and with error message. This is a BadRequest.

Reply:
maybe for `utils.ValidateCronString()` we should only return `error`, instead of return `(bool, error)`